aws/test/colored-money-hardhat/test/RLN.test.ts

```diff
@@ -37,42 +74,42 @@ describe('Test RLN functionality', async function () {
   });
 
   it('Deposits', async () => {
-    const pendingDepositsInitial = await user.checkPendingDeposits();
-    let balanceInitial = 0;
-    try {
-      balanceInitial = pendingDepositsInitial[UserConfig.CONTRACT_ADDRESS.toLowerCase()][1].balance;
-    } catch (error) {
-      // No initial pending balance
-    }
-    try {
-      // Make deposit
-      const txReceipts = await user.makeDeposit({
-        tokenContractAddress: UserConfig.CONTRACT_ADDRESS,
-        value: valueDeposit,
-        tokenId: entityId,
-        feeWei: '0',
-      });
-      // Check we have transaction hashes in Nightfall
-      expect(txReceipts.txReceipt.transactionHash).to.a('string').and.to.include('0x');
-      expect(txReceipts.txReceiptL2.transactionHash).to.a('string').and.to.include('0x');
-      expect(user.nightfallDepositTxHashes[0]).to.a('string').and.to.include('0x');
-    } catch (error) {
-      console.log('Error in deposit', error);
+    const balanceInitial = await getRLNBalance(user, UserConfig.CONTRACT_ADDRESS, entityId);
+    for (let i = 0; i < 2; i++) {
+      try {
+        // Make deposit
+        const txReceipts = await user.makeDeposit({
+          tokenContractAddress: UserConfig.CONTRACT_ADDRESS,
+          value: valueDeposit,
+          tokenId: entityId,
+          feeWei: '0',
+        });
+        // Check we have transaction hashes in Nightfall
+        expect(txReceipts.txReceipt.transactionHash).to.a('string').and.to.include('0x');
+        expect(txReceipts.txReceiptL2.transactionHash).to.a('string').and.to.include('0x');
+        expect(user.nightfallDepositTxHashes[0]).to.a('string').and.to.include('0x');
+      } catch (error) {
+        console.log('Error in deposit', error);
+      }
     }
-    const pendingDepositsFinal = await user.checkPendingDeposits();
-    const balanceFinal = pendingDepositsFinal[UserConfig.CONTRACT_ADDRESS.toLowerCase()][1].balance;
-    expect(balanceFinal - balanceInitial).to.be.equal(Number(valueDeposit));
+    const balanceFinal = await getRLNBalance(user, UserConfig.CONTRACT_ADDRESS, entityId);
+    expect(balanceFinal - balanceInitial).to.be.equal(2 * Number(valueDeposit));
   });
 
   it('Transfers', async () => {
-    const balanceNightfallInitial = await user.checkNightfallBalances();
-    let balanceInitial = 0;
-    try {
-      balanceInitial =
-        balanceNightfallInitial[UserConfig.CONTRACT_ADDRESS.toLowerCase()][0].balance;
-    } catch (error) {
-      // No initial pending balance
+    let confirmedBalance = await getRLNBalance(user, UserConfig.CONTRACT_ADDRESS, entityId, true);
+    while (confirmedBalance < 2 * Number(valueDeposit)) {
+      console.log(
+        `Waiting for confirmed balance to be > ${
+          2 * Number(valueDeposit)
+        } (${confirmedBalance})...`,
+      );
+      await new Promise(resolve => setTimeout(resolve, 10000));
```

Comment:
I am unsure why this delay is forced here?

Reply:
This is to wait for the deposit to be mined in the blockchain as a L2 block to be available in Nightfall. In this loop I'm checking this Nightfall balance to be at least enough to do the Transfer and then the Withdraw. We expect at least have the value of these 2 deposits.